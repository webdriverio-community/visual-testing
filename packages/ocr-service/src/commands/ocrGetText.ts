import ocrGetData from '../utils/ocrGetData.js'
import type { OcrGetTextOptions } from '../types.js'

export default async function ocrGetText(options: OcrGetTextOptions): Promise<string> {
    const { text } = await ocrGetData(options)

    return text.replace(/\n\s*\n/g, '\n')
}
