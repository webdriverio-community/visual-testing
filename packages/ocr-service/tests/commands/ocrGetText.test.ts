import { describe, it, expect, vi } from 'vitest'
import ocrGetText from '../../src/commands/ocrGetText.js'
import ocrGetData from '../../src/utils/ocrGetData.js'

vi.mock('../../src/utils/ocrGetData.js', () => ({
    default: vi.fn(() => Promise.resolve({
        text: `

        Next-gen browser and mobile automation

        test framework for Node.js
         `
    }))
}))

describe('ocrGetText', () => {
    it('should return text', async () => {
        const options = {
            contrast: 0.5,
            isTesseractAvailable: false,
            language: 'ENG',
            ocrImagesPath: 'path/to/image.png',
            text: 'example',
        }
        expect(await ocrGetText(options)).toMatchSnapshot()
        expect(ocrGetData).toHaveBeenCalledWith(options)
    })
})
