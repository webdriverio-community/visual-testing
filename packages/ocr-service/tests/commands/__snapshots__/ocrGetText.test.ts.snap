// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ocrGetText > should return text 1`] = `
"
        Next-gen browser and mobile automation
        test framework for Node.js
         "
`;
